Connectors/Navisworks/Speckle.Connectors.NavisworksShared/Operations/Send/GeometryNodeMerger.cs

```diff
@@ -7,16 +8,156 @@ namespace Speckle.Connector.Navisworks.Operations.Send;
+
+    // We can as many signature defining methods as we want here
+    AddMaterialProperties(node, signatureProperties);
+
+    // When we are done adding properties, we can generate the signature
+    return GetSignature(signatureProperties);
+  }
+
+  /// <summary>
+  /// Adds material-related properties to the properties dictionary
+  /// </summary>
+  private static void AddMaterialProperties(NAV.ModelItem node, Dictionary<string, object> properties)
+  {
+    if (!node.HasGeometry || node.Geometry == null)
+    {
+      return;
+    }
+
+    var geometry = node.Geometry;
+    if (geometry.ActiveColor != null)
+    {
+      properties["ActiveColor"] = (geometry.ActiveColor.R, geometry.ActiveColor.G, geometry.ActiveColor.B);
+      properties["ActiveTransparency"] = geometry.ActiveTransparency;
+    }
+
+    // Add material name if available
+    var materialName = GetMaterialName(node);
+    if (!string.IsNullOrEmpty(materialName))
+    {
+      properties["MaterialName"] = materialName;
+    }
+  }
+
+  /// <summary>
+  /// Creates a hash-based signature from a dictionary of properties.
+  /// </summary>
+  /// <param name="properties">Dictionary containing property name/value pairs to include in the signature</param>
+  /// <param name="hashLength">Length of the returned hash string (default: 8 characters)</param>
+  /// <returns>A hash string representing the combined properties</returns>
+  private static string GetSignature(Dictionary<string, object> properties, int hashLength = 8)
+  {
+    if (properties.Count == 0)
+    {
+      return "empty";
+    }
+
+    // Build a consistent string representation of all properties
+    var hashInput = new System.Text.StringBuilder();
+
+    // Sort keys to ensure consistent order
+    var sortedKeys = properties.Keys.OrderBy(k => k).ToList();
+
+    foreach (var key in sortedKeys)
+    {
+      var value = properties[key];
+      switch (value)
+      {
+        case null:
+          continue;
+        // Format numbers with fixed precision to avoid floating point inconsistencies
+        case double doubleValue:
+          hashInput.Append($"{key}:{Math.Round(doubleValue, 6)}_");
+          break;
+        case float floatValue:
+          hashInput.Append($"{key}:{Math.Round(floatValue, 6)}_");
+          break;
+        default:
+          hashInput.Append($"{key}:{value.GetHashCode()}_");
+          break;
+      }
+    }
+
+    return hashInput.Length == 0 ? "empty" : GetHash(hashInput.ToString(), hashLength);
+  }
+
+  /// <summary>
+  /// Generates a non-cryptographic hash from the input string. Mostly to avoid MD5.Create() warnings.
+  /// </summary>
+  /// <param name="input">The input string to hash.</param>
+  /// <param name="length">The length of the returned hash string (default: 8 characters).</param>
+  /// <returns>A non-cryptographic hash string of the specified length.</returns>
+  private static string GetHash(string input, int length = 8) =>
```

Comment:
Don't do this...use MD5 if you mean to or a SHA version

Reply:
Fair. I started with MD5.Create() and got security flagged. It's not needed to be tremendously secure 🫠

Reply:
Can just pragma around it if it doesn't need to be secure and MD5 is good enough